src/components/SecondaryPanes/Scopes.js

```diff
@@ -59,27 +66,45 @@ class Scopes extends PureComponent<Props, State> {
           nextProps.why,
           nextProps.selectedFrame,
           nextProps.frameScopes
+        ),
+        generatedScopes: getScopes(
+          nextProps.why,
+          nextProps.selectedFrame,
+          nextProps.generatedFrameScopes
         )
       });
     }
   }
 
   render() {
-    const { isPaused, isLoading } = this.props;
-    const { scopes } = this.state;
+    const { hasOriginalScopes, isPaused, isLoading } = this.props;
+    const { scopes, generatedScopes, showOriginal } = this.state;
 
     if (scopes) {
       return (
         <div className="pane scopes-list">
           <ObjectInspector
-            roots={scopes}
+            roots={showOriginal ? scopes : generatedScopes}
             autoExpandAll={false}
             autoExpandDepth={1}
             disableWrap={true}
             disabledFocus={true}
             dimTopLevelWindow={true}
             createObjectClient={grip => createObjectClient(grip)}
           />
+          {hasOriginalScopes ? (
+            <div className="scope-type-toggle">
+              <a
+                href=""
+                onClick={e => {
+                  e.preventDefault();
+                  this.setState({ showOriginal: !showOriginal });
+                }}
+              >
+                Show {showOriginal ? "generated" : "original"} scope
```

Comment:
hmm, i think we should have two strings in `assets/panel/debugger.properties` not a big deal... See 115 for the L10.getStr api

Reply:
Sounds good.